<?xml version="1.0" encoding="UTF-8" ?>

<launch>
    <arg name="device" default="cuda" doc="Device to run tensor operations on: cpu or cuda"/>
    <arg name="weights" default="fcn_resnet101_lr_0.0001_bs_4_epoch_5_TraversabilityClouds_depth_labels_traversability_iou_0.636.pth"
         doc="Point cloud segmentation network weights file (located in '../config/weights/depth_cloud/' folder)"/>
    <arg name="cloud_in" default="/os_cloud_node/points" doc="Input point cloud topic"/>

    <node name="cloud_segmentation_inference" pkg="traversability_estimation" type="cloud_segmentation_inference" output="screen">
        <env name="PYTHONPATH" value="/home/robot/workspace/traversability/src/traversability_estimation/thirdparty/vision:$(optenv PYTHONPATH)"/>
        <rosparam subst_value="true">
            device: $(arg device)
            max_age: 0.2
            lidar_channels: 128
            lidar_beams: 1024
            lidar_fov_up: 45.0
            lidar_fov_down: -45.0
            input_fields: ['x', 'y', 'z', 'intensity']
            output_fields: ['x', 'y', 'z', 'r', 'g', 'b', 'obstacle', 'obstacle_soft']
            labels_to_correct: []
            weights: $(arg weights)
        </rosparam>
        <rosparam param="label_map" command="load" file="$(dirname)/../config/rellis_to_traversability.yaml"/>

        <remap from="cloud_in" to="$(arg cloud_in)"/>
        <remap from="cloud_out" to="/cloud_segmentation/lidar_cloud"/>
    </node>

</launch>
